### PR TITLE
Add documentation for logging from extensions

### PR DESCRIPTION
**Pull request by robertpanzer**
Thank you for opening a pull request and contributing to AsciidoctorJ!

Please take a bit of time giving some details about your pull request:

## Kind of change

- [ ] Bug fix
- [ ] New non-breaking feature
- [ ] New breaking feature
- [x] Documentation update
- [ ] Build improvement

## Description

Add documentation for how an extension can log messages that are also captured by build tools like the maven or gradle plugin.

### CODE REVIEW

File: docs/modules/ROOT/pages/asciidoctor-api-options.adoc [hunk 1]
```diff
@@ -115,6 +115,19 @@ OptionsBuilder optionsBuilder =
             Options.builder().templateDirs(new File("templates_path"));
 ----
 
+[[sourcemap]]
+== sourcemap
+
+Keeps track of the file and line number for each parsed block. (Useful for tooling applications where the association between the converted output and the source file is important).
```
Comment (abelsromero) [inline docs/modules/ROOT/pages/asciidoctor-api-options.adoc:121]: I think the parenthesis are a leftover?

**Comment by mojavelinux** [comment]
I tested this and it works great. This is exactly what I had in mind for being able to log from an extension.

```
log(new org.asciidoctor.log.LogRecord(org.asciidoctor.log.Severity.WARN, "Hello, World!"));
```

**Comment by mojavelinux** [comment]
One thing I thing we should add to the docs is how to configure JUL so that it matches the default log output of Asciidoctor Ruby. This is more of demonstration, but it gives the user a launching off point.

logging.properties
```
handlers = java.util.logging.ConsoleHandler
.level = WARNING
java.util.logging.ConsoleHandler.formatter = java.util.logging.SimpleFormatter
java.util.logging.SimpleFormatter.format = %3$
```

Calling asciidoctorj:

```
JAVA_OPTS="-Djava.util.logging.config.file=$PWD/logging.properties" asciidoctorj document.adoc
```

Let me know if you'd like a separate issue for that and I'll file it.

**Comment by mojavelinux** [comment]
Keep in mind that if the default level is INFO instead of WARNING, users will see provisional messages about possible invalid xrefs...which could be valid xrefs that Asciidoctor just can't resolve.

File: docs/modules/ROOT/pages/asciidoctor-api-options.adoc (hunk 1 above)
Comment (robertpanzer) [reply docs/modules/ROOT/pages/asciidoctor-api-options.adoc:121]: I copied this sentence as is from the Asciidoctor.js module to make sure that we don't provide deviating definitions. :D
But yeah, I'll remove the parens.

**Comment by robertpanzer** [comment]
Thanks for the feedback!
I added some notes on this to the description of the log handling API, since this is also the place that mentions that log messages are automatically forwarded to JUL.

**Comment by robertpanzer** [comment]
Added a comment to the converter docs that they can also log by calling `log()`.
The current docs "just" show how to implement a string based converter that extends the abstract class StringConverter.
For arbitrary, binary formats there is also an approach based exclusively on interfaces.
We might want to document this in the future.

**Pull request merged**